src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/TokenCertificateService.java

```diff
@@ -372,6 +382,83 @@ private ClientId getClientIdForSigningCert(String instanceIdentifier, X509Certif
         return certificateSubject;
     }
 
+    /**
+     * Send the authentication certificate registration request to central server
+     * @param hash certificate hash
+     * @param securityServerAddress IP address or DNS name of the security server
+     * @throws CertificateNotFoundException
+     * @throws GlobalConfService.GlobalConfOutdatedException
+     */
+    public void registerAuthCert(String hash, String securityServerAddress) throws CertificateNotFoundException,
+            GlobalConfService.GlobalConfOutdatedException, InvalidCertificateException,
+            SignCertificateNotSupportedException {
+        CertificateInfo certificateInfo = getCertificateInfo(hash);
+        verifyAuthCert(certificateInfo);
+        SecurityServerId securityServerId = serverConfService.getSecurityServerId();
+        try {
+            managementRequestService.sendAuthCertRegisterRequest(securityServerId, securityServerAddress,
+                    certificateInfo.getCertificateBytes());
+            signerProxyFacade.setCertStatus(certificateInfo.getId(), CertificateInfo.STATUS_REGINPROG);
+        } catch (GlobalConfService.GlobalConfOutdatedException e) {
+            throw e;
+        } catch (Exception e) {
+            if (e instanceof CodedException) {
+                CodedException ce = (CodedException) e;
+                if (isCausedByCertNotFound(ce)) {
```

Comment:
Since getCertificateInfo already throws CertificateNotFoundException, there's a possibility to simplify exception handling by skipping isCausedByCertNotFound here.

Reply:
True. And furthermore if `getCertificateInfo` finds the cert and the latter method logic doesn't -> it is more likely a runtime error at that point anyway.